Backend/Controllers/TransactionsController.cs

```diff
@@ -16,44 +22,84 @@ namespace Backend.Controllers
     [ApiController]
     public class TransactionsController : ApiControllerBase
     {
-        public TransactionsController(ApiContext context, UserManager<User> userManager) : base(context, userManager) { }
+        public TransactionsController(ApiContext context, UserManager<User> userManager) :
+            base(context, userManager) { }
 
         [HttpPost]
+        [Authorize(Roles = "Pharmacy")]
         public async Task<IActionResult> CreateTransaction([FromBody] CreateTransactionDTO dto)
+        {
+            try
+            {
+                var user = await GetCurrentUser();
+
+                var pharmacy = await RetrievePharmacy(user.PharmacyId);
+                var register = RetrieveRegister(pharmacy, dto.RegisterId);
+
+                var transaction = new Transaction(pharmacy.Id, dto);
+
+                foreach (var product in dto.Products)
+                {
+                    TryAddProduct(pharmacy, product, transaction);
+                }
+
+                if (transaction.PaymentTypeId == PaymentTypeId.Cash)
+                {
+                    register.Cash += transaction.TotalPrice;
+                }
+
+                Context.Add(transaction);
+                await Context.SaveChangesAsync();
+
+                return Created();
+            }
+            catch (ResourceNotFoundException ex)
+            {
+                return ApiNotFound(ApiErrorSlug.ResourceNotFound, ex.Message);
+            }
+            catch (ArgumentException ex)
+            {
+                return ApiBadRequest(ApiErrorSlug.InsufficientBalance, ex.Message);
+            }
+        }
+
+        private static void TryAddProduct(Pharmacy pharmacy, TransactionProductDTO dto, Transaction transaction)
+        {
+            var productInPharmacy = pharmacy.Products.FirstOrDefault(pb => pb.Id == dto.ProductBalanceId);
+
+            if (productInPharmacy == null) throw new ResourceNotFoundException("productBalance");
+            if (productInPharmacy.Amount < dto.Amount)
+            {
+                throw new ArgumentException("amount");
+            }
+
+            productInPharmacy.Amount -= dto.Amount;
+            // TODO: Check amount and append to the order of required
```

Comment:
Is it done?

Reply:
Not yet, it will be implemented with the next MR. This is the placeholder, where the method should be called